File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        stack<char> st;
        string ans="";
        for(int i=0;i<s.length();i++){
            if(st.size()==0){
                st.push(s[i]);
            }
            else{
                if(s[i]=='('){
                    ans+=s[i];
                    st.push(s[i]);
                }
                else if(s[i]==')'){
                    if(st.size()) st.pop();
                    if(st.size()!=0){
                        ans+=s[i];
                    }
                }
            }
        }
        return ans;
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cs
public class Solution {
    public string RemoveOuterParentheses(string l) {
        StringBuilder sb=new StringBuilder(l);
        int idx=sb.Length-1;
        int count=0;
        for(int i=sb.Length-1;i>=0;i--){
            if(sb[i]==')') count++;
            else{
                count--;
            }
            if(count==0){
                sb.Remove(idx,1);
                sb.Remove(i, 1);
                idx=i-1;
                count=0;
            }
        }
        return sb.ToString();

    }
}